app/domain/references/service.py

```diff
@@ -922,6 +936,28 @@ async def create_robot_enhancement_batch(
             reference_data_file=reference_data_file,
         )
 
+    @sql_unit_of_work
+    async def renew_robot_enhancement_batch_lease(
+        self,
+        robot_enhancement_batch_id: UUID,
+        lease_duration: datetime.timedelta,
+    ) -> datetime.datetime:
+        """Renew a robot enhancement batch lease."""
+        await self.sql_uow.robot_enhancement_batches.verify_pk_existence(
+            [robot_enhancement_batch_id]
+        )
+        expiry = apply_positive_timedelta(lease_duration)
+        await self.sql_uow.pending_enhancements.bulk_update_by_filter(
+            filter_conditions={
+                "robot_enhancement_batch_id": robot_enhancement_batch_id,
+                # If a robot lets a pending enhancement expire, it must use a
+                # new robot enhancement batch to re-process it.
+                "status": PendingEnhancementStatus.PROCESSING,
```

Comment:
This will fail silently if no pending enhancements are renewed right? I wonder if we should throw an error in that case.